plugins/modules/kutils.py

```diff
@@ -211,27 +207,21 @@ def __init__(self, module, **kwargs):
 
         self.kdestroy = CmdRunner(
             module,
-            command=self.kdestroy_executable,
+            command='kdestroy',
             arg_formats=dict(
                 kdestroy_all=cmd_runner_fmt.as_bool('-A'),
                 cache_name=cmd_runner_fmt.as_opt_val('-c'),
                 principal=cmd_runner_fmt.as_opt_val('-p'),
             )
         )
 
-    def check_for_none(self, param, runner_arg):
-        """
-        param is a bool, runner_arg is a string argument (e.g. -a)
-        If param=True  -> return -f/-p/-a
-        If param=False -> return -F/-P/-A
-        if param=None  -> return []
-        """
-        if param is None:
-            return []
-        if param:
-            return runner_arg.lower()
-        else:
-            return runner_arg
+        self.klist = CmdRunner(
+            module,
+            command='klist',
+            arg_formats=dict(
+                list=cmd_runner_fmt.as_bool('-l'),
```

Comment:
I would strongly recommend you do not use `list`, so that it does not clash the Python builtin class. Same below where you actually use the runner.